src/mesh.cpp

```diff
@@ -236,6 +236,63 @@ vector<Mesh::MaterialVolume> Mesh::material_volumes(
   return result;
 }
 
+void Mesh::sum_distance_xs(const Position& start, const Position& end) const
+{
+  Position direction = {end.x - start.x, end.y - start.y, end.z - start.z};
+  double magnitude =
+    std::sqrt(direction.x * direction.x + direction.y * direction.y +
+              direction.z * direction.z);
+
+  // Normalize the direction
+  direction.x /= magnitude;
+  direction.y /= magnitude;
+  direction.z /= magnitude;
+
+  Particle p;
+  SourceSite site;
+  site.E = 1.0;
+  site.particle = ParticleType::neutron;
+  site.r = start;
+  site.u = direction;
+  p.from_source(&site);
+
+  double total_distance_xs = 0.0;
```

Comment:
this is the optical depth or optical thickness, so let's call it that....
```suggestion
  double optical_thickness = 0.0;
```